src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -490,13 +490,12 @@ private IEnumerable<Location> GetDiagnosticLocationsWithinTree(SyntaxTree tree)
             }
         }
 
-        internal Diagnostic WithReportDiagnostic(ReportDiagnostic reportAction)
+        internal Diagnostic WithReportDiagnostic(ReportDiagnostic reportAction, bool reportSuppressedDiagnostics)
         {
             switch (reportAction)
```

Comment:
>switch (reportAction) [](start = 12, length = 21)

Before getting into this ```switch```, should we do something special about Diagnostic that is already IsSuppressed?